crates/gitbutler-branch-actions/src/stack.rs

```diff
@@ -226,12 +233,14 @@ fn series_integrated(check_commit: &IsCommitIntegrated, series: &Series) -> Resu
 
 /// Returns the stack series for the API.
 /// Newest first, oldest last in the list
+/// `commits` is used to accelerate the is-integrated check.
 pub(crate) fn stack_series(
     ctx: &CommandContext,
     branch: &mut Stack,
     default_target: &Target,
-    check_commit: &IsCommitIntegrated,
+    check_commit: &mut IsCommitIntegrated,
     remote_commit_data: HashMap<CommitData, git2::Oid>,
+    commits: &[VirtualBranchCommit],
```

Comment:
In general this is probably fine - but the commits that are passed in here, are something deprecated that I plan to remove asap

Reply:
I am looking forward to seeing that work only done once, and what I understood is that `stack_series` is the new version, and I was basically altering the old one. The few changes I made should be easy to port to be internal to `stack_series()` (if this is what's needed).